src/overlays/actors/ovl_Bg_Jya_1flift/z_bg_jya_1flift.c

```diff
@@ -21,31 +37,164 @@ const ActorInit Bg_Jya_1flift_InitVars = {
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_1flift/BgJya1flift_Draw.s")
+extern UNK_TYPE D_060004A8;
+extern Gfx D_060001F0[];
+
+void BgJya1flift_InitDynapoly(BgJya1flift* this, GlobalContext* globalCtx, UNK_PTR arg2, s32 moveFlag) {
+    s32 pad;
+    s32 localConst;
+    s32 pad2;
+
+    localConst = 0;
+    DynaPolyInfo_SetActorMove(&this->dyna, moveFlag);
+    DynaPolyInfo_Alloc(arg2, &localConst);
+    this->dyna.dynaPolyId =
+        DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, localConst);
+
+    if (this->dyna.dynaPolyId == 0x32) {
+        // Warning : move BG login failed
+        osSyncPrintf("Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_bg_jya_1flift.c", 0xB3,
+                     this->dyna.actor.id, this->dyna.actor.params);
+    }
+}
+
+void BgJya1flift_InitCollision(Actor* thisx, GlobalContext* globalCtx) {
+    BgJya1flift* this = THIS;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+    this->dyna.actor.colChkInfo.mass = 0xFF;
+}
+
+void BgJya1flift_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgJya1flift* this = THIS;
+    // 1 F lift
+    osSyncPrintf("(１Ｆリフト)(flag %d)(room %d)\n", D_808930E0, globalCtx->roomCtx.curRoom.num);
+    this->hasInitialized = 0;
+    if (D_808930E0 != 0) {
+        Actor_Kill(thisx);
+        return;
+    }
+    BgJya1flift_InitDynapoly(this, globalCtx, &D_060004A8, 0);
+    Actor_ProcessInitChain(thisx, sInitChain);
+    BgJya1flift_InitCollision(thisx, globalCtx);
+    if (Flags_GetSwitch(globalCtx, (thisx->params & 0x3F)) != 0) {
+        LINK_AGE_IN_YEARS == YEARS_ADULT ? BgJya1flift_ChangeDirection(this) : func_80892E0C(this);
+    } else {
+        func_80892DB0(this);
+    }
+    thisx->room = -1;
+    D_808930E0 = 1;
+    this->hasInitialized = 1;
+}
+
+void BgJya1flift_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgJya1flift* this = THIS;
+
+    if (this->hasInitialized != 0) {
+        D_808930E0 = 0;
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+        DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    }
+}
+
+void func_80892DB0(BgJya1flift* this) {
+    this->actionFunc = func_80892DCC;
+    this->dyna.actor.posRot.pos.y = finalPositions[0];
+}
+
+void func_80892DCC(BgJya1flift* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, (this->dyna.actor.params & 0x3F))) {
+        BgJya1flift_ChangeDirection(this);
+    }
+}
+
+void func_80892E0C(BgJya1flift* this) {
+    this->actionFunc = BgJya1flift_DoNothing;
+    this->dyna.actor.posRot.pos.y = finalPositions[0];
+}
+
+void BgJya1flift_DoNothing(BgJya1flift* this, GlobalContext* globalCtx) {
+}
+
+void BgJya1flift_ChangeDirection(BgJya1flift* this) {
+    this->actionFunc = BgJya1flift_Move;
+    this->direction ^= 1;
+    this->dyna.actor.velocity.y = 0.0f;
```

Comment:
i guess my suggestion of making this a boolean doesnt make much sense for xor? not really sure how that would work here. would like to hear other input

Reply:
It makes sense to keep the XOR since XORing just flips it anyway.  XOR is also whats in the ASM anyway.  Treating the variable as a bool should still work though.